### PR TITLE
executor: Modify temporary table behavior for history read

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Problem Summary:

Some tools like dumpling or mydumper will read all tables to dump data. To make a compatible behavior with these tools, a history read to temp table should not return error

### What is changed and how it works?

do not return error when history read temporary table

### Related changes

- N/A

### Check List 

Tests 

- Unit test

Side effects

- N/A

### Release note 

- No release note

### CODE REVIEW

**Comment by tiancaiamao** [comment]
ref https://github.com/pingcap/tidb/pull/25724

File: executor/executor_test.go [hunk 1]
```diff
@@ -8652,7 +8652,8 @@ func (s *testStaleTxnSuite) TestInvalidReadTemporaryTable(c *C) {
 
 	tk.MustExec("set @@tidb_snapshot=NOW(6)")
 	for _, query := range queries {
-		tk.MustGetErrMsg(query.sql, "can not read temporary table when 'tidb_snapshot' is set")
+		// Will success here for compatibility with some tools like dumping
+		tk.MustQuery(query.sql)
```
Comment (djshow832) [inline executor/executor_test.go:8656]: Check the result.

**Review by djshow832 — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Comment by tiancaiamao** [comment]
We will need to update the document later about the behaviour

**Comment by djshow832** [comment]
> We will need to update the document later about the behaviour

@tiancaiamao Which document? The RFC is fine,  it's normal that it's out of date. The user doc is not filed.

**Pull request merged**